### PR TITLE
Alpine 3.16

### PR DESCRIPTION
**Pull request by giggsoff**
Let's update eve-alpine to be based on Alpine 3.16. Update for new-kernel, acrn and acrn-kernel removed from this PR to fit timeout of CI.

### CODE REVIEW

File: .github/workflows/publish.yml
Comment (eriknordmark) [inline .github/workflows/publish.yml:47]: Why 3.13 here?

File: pkg/alpine/Dockerfile
Comment (eriknordmark) [inline pkg/alpine/Dockerfile:1]: Just to make sure I understand the flow, once we have build a new lfedge/eve-alpine using this PR, will we change this line to refer to lfedge/eve-alpine:<version>? 
I assume we need that to ensure reproducible builds.

File: pkg/alpine/eve-alpine-deploy.sh
Comment (eriknordmark) [inline pkg/alpine/eve-alpine-deploy.sh:19]: This is an example, but why not use 3.18?

**Review by eriknordmark — APPROVED** [review]
Let it test.

File: .github/workflows/publish.yml
Comment (giggsoff) [reply .github/workflows/publish.yml]: riscv64 packages was not ready, but now they are, so I updated PR to build riscv64 EVE as well on top of Alpine 3.16

File: pkg/alpine/Dockerfile
Comment (giggsoff) [reply pkg/alpine/Dockerfile:1]: Correct, also I will add new-kernel, acrn and acrn-kernel updates (with them we are not fit CI timeout for arm64).

**Comment by giggsoff** [comment]
> Let it test.

I hope I [solve](https://github.com/lf-edge/eve/pull/2780/files#diff-ac769ff4dfca87afdd5a9c23b38a31c37d2b4f1b854e764d8e0c50ad05d086a7R79) the issue with old test's trial failed. It was because of patch for dhcpcd ends with two files (new one and another file with `.old` suffix and without changes) and we had wrong `/etc/resolv.conf` file.

**Comment by eriknordmark** [comment]
@giggsoff some arm64 issue:
ERROR: unable to select packages:
  binutils-gold (no such package):

**Review by eriknordmark — APPROVED** [review]
Run eden

File: pkg/alpine/Dockerfile
Comment (eriknordmark) [inline pkg/alpine/Dockerfile:26]: hadolint complains here

File: pkg/alpine/mirrors/3.16/community
Comment (eriknordmark) [inline pkg/alpine/mirrors/3.16/community:5]: I'm a bit concerned about depending on another ssl/crypto library. Which package(s) require libresll? Shouldn't openssl suffice?

File: pkg/alpine/mirrors/3.16/main
Comment (eriknordmark) [inline pkg/alpine/mirrors/3.16/main:91]: Also a TLS package?

File: pkg/debug/Dockerfile
Comment (eriknordmark) [inline pkg/debug/Dockerfile:6]: Is there a way to define this in a common place which all the pkg/*/Dockerfile can include?

Comment (giggsoff) [reply pkg/debug/Dockerfile:6]: We had templated Dockerfile.in before where we can put something like ALPINE_TAG here, but decided to get rid of them (it was discussion here for example https://github.com/lf-edge/eve/pull/2353). So now it is not easy. Actually I also tired to rebase all commits of the PR on top of updated eve-alpine, but it helps us to have clear Dockerfiles which we can use without any templated magic around.
Strictly speaking support for build args are not upstreamed into linuxkit and we use them only for risc now with hacked publish workflow. We use build-args there to force build for risc (we force risc arch in base image so the whole image builded for risc arch as well). But with updated linuxkit we work on in the separate PR we can remove build arg here and just use simple FROM.

File: pkg/alpine/mirrors/3.16/community
Comment (giggsoff) [reply pkg/alpine/mirrors/3.16/community:5]: Will check. Right now those lines in diff are mostly about moving from main to comminity repo.

**Comment by giggsoff** [comment]
> @giggsoff some arm64 issue: ERROR: unable to select packages: binutils-gold (no such package):

This package is not available for risc. Put additional logic to have separate subset of packages per arch.
Also seems we have another problem with Eden workflow, which was failed again. We try to use pushed evedanger image, but it was not update of the image in docker because of error in arm64 build. In general in case of the second approve after some PR modification come during build of new version we will re-use the first published evedanger image. It is wrong, will try to fix in separate PR.

File: pkg/alpine/mirrors/3.16/community
Comment (giggsoff) [reply pkg/alpine/mirrors/3.16/community:5]: I can see that we use it here https://github.com/lf-edge/eve/blob/0c6de671a3a66ede7a3ea2d417397d08b7897a6f/pkg/xen/Dockerfile#L5
and here
https://github.com/lf-edge/eve/blob/0c6de671a3a66ede7a3ea2d417397d08b7897a6f/pkg/acrn-kernel/Dockerfile#L5
I can try to build them with openssl if you suggest.

File: pkg/alpine/mirrors/3.16/main
Comment (giggsoff) [reply pkg/alpine/mirrors/3.16/main:91]: Well, seems it is not required anywhere now, will try to rebuild without this library.

File: pkg/alpine/mirrors/3.16/community
Comment (eriknordmark) [reply pkg/alpine/mirrors/3.16/community:5]: I see we already have it (my eyes might make missed it was a move).
So we can explore replacing it with openssl in a separate PR.

File: pkg/debug/Dockerfile
Comment (eriknordmark) [reply pkg/debug/Dockerfile:6]: OK

**Comment by mikem-zed** [comment]
Why do we do it in the first place? Why we need Alpine 3.16 ?

**Comment by giggsoff** [comment]
> Why do we do it in the first place? Why we need Alpine 3.16 ?

It was inspired by another PRs: https://github.com/lf-edge/eve/pull/2758 and https://github.com/lf-edge/eve/pull/2762. We observed that if we want to update go, it would be nice to update all packages as well https://github.com/lf-edge/eve/pull/2762#issuecomment-1215268475.

File: pkg/alpine/mirrors/3.16/main
Comment (giggsoff) [reply pkg/alpine/mirrors/3.16/main]: Removed

File: pkg/alpine/Dockerfile
Comment (giggsoff) [reply pkg/alpine/Dockerfile]: fixed

**Comment by eriknordmark** [comment]
> Small comment here: after merging of #2727 we will not be able to use the logic in publish workflow from this PR (after merging of #2727 linuxkit will use images from own cache, not from docker, so we will not be able to bootstrap base image using docker commands).

@giggsoff @deitch What is the preferred order between #2727 and this #2780?

**Comment by giggsoff** [comment]
> > Small comment here: after merging of #2727 we will not be able to use the logic in publish workflow from this PR (after merging of #2727 linuxkit will use images from own cache, not from docker, so we will not be able to bootstrap base image using docker commands).
> 
> @giggsoff @deitch What is the preferred order between #2727 and this #2780?

This one first, then next (future) PR with update of tag in pkg/eve-alpine, then #2727. This is the simplest option, but if there is a different priority, changes will be needed in this PR (and not yet obvious to me).

**Comment by eriknordmark** [comment]
@giggsoff Should we fix the hadolint: DL4006 complaint or is that something we can ignore?

**Comment by giggsoff** [comment]
> @giggsoff Should we fix the hadolint: DL4006 complaint or is that something we can ignore?

I can add comment with ignore there or we can ignore it for now as soon we may delete build-args as useless after linuxkit update.

**Comment by giggsoff** [comment]
Well, if we merged #2727 I will mark this as WIP for now as several updates required.

**Comment by deitch** [comment]
I recommend rebasing now that #2727 is in

**Comment by giggsoff** [comment]
I rebased on top of linuxkit update. Also as we do not support build args now I was forced to not use `FROM alpine:3.16.2` in eve-alpine (as there is no such image for riscv64) and add another package eve-alpine-base as bootstrap image instead.

**Review by eriknordmark — APPROVED** [review]
Re-run eden

**Comment by giggsoff** [comment]
Seems, after merging of containerd/runc update I am forced to rebase Alpine 3.16 PR again because of pillar’s Dockerfile require update (because of dom0-tools hash updated), sorry. Feel free to cancel/restart PR build workflow if you will need builder for something urgent (it will consume about 5 hours to build).

**Review by eriknordmark — APPROVED** [review]
Re-run eden

**Pull request merged**
